### PR TITLE
tftui: init at 0.12.4

### PR DESCRIPTION
**Pull request by cdepillabout**
## Description of changes

[`tftui`](https://github.com/idoavrah/terraform-tui) is a TUI tool for viewing Terraform state files:

https://github.com/idoavrah/terraform-tui/blob/main/demo/tftui.gif

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/cluster/tftui/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+, requests
+, rich
+, textual
+, enableUsageTracking ? false
+}:
+
+buildPythonApplication rec {
+  pname = "tftui";
+  version = "0.12.4";
+  format = "pyproject";
```
Comment (zeuner) [inline pkgs/applications/networking/cluster/tftui/default.nix:17]: Should be `pyproject = true;` now (see https://github.com/NixOS/nixpkgs/commit/39c64e232328e09284c766d2b56f2d680e511d41).

File: pkgs/applications/networking/cluster/tftui/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      --add-flags "--disable-usage-tracking"
+  '';
+
+  meta = with lib; {
+    description = "Textual UI to view and interact with Terraform state";
+    homepage = "https://github.com/idoavrah/terraform-tui";
+    changelog = "https://github.com/idoavrah/terraform-tui/releases";
+    license = licenses.asl20;
+    maintainers = with maintainers; teams.bitnomial.members;
+  };
```
Comment (zeuner) [inline pkgs/applications/networking/cluster/tftui/default.nix:52]: Please consider also setting a `meta.mainProgram` so the package can be invoked using `nix run`.

**Review by zeuner — COMMENTED** [review]
Also, according to PyPI, at least Python 3.9 is required, which should be reflected in the package (e.g. using `disabled = pythonOlder "3.9";`) somewhere.

File: pkgs/applications/networking/cluster/tftui/default.nix (hunk 2 above)
Comment (cdepillabout) [reply pkgs/applications/networking/cluster/tftui/default.nix:53]: Thanks for the review.  I added `meta.mainProgram` in 795f3e1cbaae and confirmed it worked with `nix run`.

(I also opened https://github.com/NixOS/nixpkgs/issues/281276 to help clear up exactly when setting `meta.mainProgram` is necessary.)

File: pkgs/applications/networking/cluster/tftui/default.nix (hunk 1 above)
Comment (cdepillabout) [reply pkgs/applications/networking/cluster/tftui/default.nix]: Thanks for the tip, fixed in 795f3e1cbaae

**Review by clf-cklf — APPROVED** [review]

**Review by wraithm — APPROVED** [review]

**Pull request merged**